### PR TITLE
Docs:  Docstrings for endpoints

### PR DESCRIPTION
**Pull request by pbullhove**
## What does this pull request change?

This is a part of "Den store Dokumentasjonsdagen". 
Adding docstrings to DMSS endpoints. 

## Issues related to this change:
https://github.com/orgs/equinor/projects/189?pane=issue&itemId=35829553

### CODE REVIEW

**Review by KristianKjerstad — COMMENTED** [review]
Good start!
However, in the API docs is a bit hard to read the argument list.
image

Could we change the syntax slightly from
```python
    """Creates a Recipe Lookup Table for an Application, given a Package Containing RecipeLinks.

    This endpoint creates a lookup table for an application. This lookup table is used to find UI- and Storage recipes given a blueprint.
    This recipe is associated with an application, based on application name.

    Args:
        application (str): Name of an application.
        recipe_package (list[str]): A list of one or more paths to packages that contain recipe links.
            Example: ["system/SIMOS/recipe_links"]
        user (User): The authenticated user accessing the endpoint.

    Returns:
        None, with status Code 204 (No Content).
    """
```

To 
```python
    """Creates a Recipe Lookup Table for an Application, given a Package Containing RecipeLinks.

    This endpoint creates a lookup table for an application. This lookup table is used to find UI- and Storage recipes given a blueprint.
    This recipe is associated with an application, based on application name.

    Args:
    - application (str): Name of an application.
    - recipe_package (list[str]): A list of one or more paths to packages that contain recipe links.
        Example: ["system/SIMOS/recipe_links"]
    - user (User): The authenticated user accessing the endpoint.

    Returns:
    - None, with status Code 204 (No Content).
    """
```


image

**Comment by ingeridhellen** [comment]
> Good start! However, in the API docs is a bit hard to read the argument list. image
> 
> Could we change the syntax slightly from
> 
> ```python
>     """Creates a Recipe Lookup Table for an Application, given a Package Containing RecipeLinks.
> 
>     This endpoint creates a lookup table for an application. This lookup table is used to find UI- and Storage recipes given a blueprint.
>     This recipe is associated with an application, based on application name.
> 
>     Args:
>         application (str): Name of an application.
>         recipe_package (list[str]): A list of one or more paths to packages that contain recipe links.
>             Example: ["system/SIMOS/recipe_links"]
>         user (User): The authenticated user accessing the endpoint.
> 
>     Returns:
>         None, with status Code 204 (No Content).
>     """
> ```
> 
> To
> 
> ```python
>     """Creates a Recipe Lookup Table for an Application, given a Package Containing RecipeLinks.
> 
>     This endpoint creates a lookup table for an application. This lookup table is used to find UI- and Storage recipes given a blueprint.
>     This recipe is associated with an application, based on application name.
> 
>     Args:
>     - application (str): Name of an application.
>     - recipe_package (list[str]): A list of one or more paths to packages that contain recipe links.
>         Example: ["system/SIMOS/recipe_links"]
>     - user (User): The authenticated user accessing the endpoint.
> 
>     Returns:
>     - None, with status Code 204 (No Content).
>     """
> ```
> 
> image

Currently dmss uses fastapi 0.78.0, while the most recent release is 0.101.1. Upgrading fastapi might solve this

File: src/features/file/file_feature.py [hunk 1]
```diff
@@ -19,8 +19,16 @@ async def upload_file(
...
     user: User = Depends(auth_w_jwt_or_pat),
 ):
-    """Upload a new binary file and create a file entity with the binary data as content.
+    """Upload a New Binary File
 
-    **file_id** The data source ID to be used for the file entity that will be created.
+    This endpoint uploads a new file and creates a file entity with the uploaded binary data as content.
+
+    Args:
+    - data_source_id (str):
```
Comment (KristianKjerstad) [inline src/features/file/file_feature.py:27]:
```suggestion
    - data_source_id (str): id to the data source to which the file should be uploaded
```

File: src/features/health_check/health_check_feature.py [hunk 2]
```diff
@@ -12,5 +12,15 @@
...
 )
 async def get():
-    """Healthcheck endpoint. Responds with "OK" - 200."""
+    """Get the Health Status Of the Service.
+
+    This endpoint can be used to check the health status of the service.
+    It always returns a 200 OK response to indicate that the service is up and running.
+
+    Args:
+    - user (User): The authenticated user accessing the endpoint.
```
Comment (KristianKjerstad) [inline src/features/health_check/health_check_feature.py:21]:
```suggestion
```
Args can probably be removed, does not look like user is used here

File: src/features/file/file_feature.py [hunk 3]
```diff
@@ -19,8 +19,16 @@ async def upload_file(
...
+    - data (dict with a "file_id" attribute): A dict containing data source ID to be used for the file entity that will be created.
+    - user (User): The authenticated user accessing the endpoint.
```
Comment (KristianKjerstad) [inline src/features/file/file_feature.py:29]:
```suggestion
    - user (User): The authenticated user accessing the endpoint, automatically generated from provided bearer token or Access-Key.
```

might be nice to include this info - can probably add this all places the user argument is used

File: src/features/lookup_table/lookup_table_feature.py [hunk 4]
```diff
@@ -41,9 +47,15 @@ def create_lookup(application: str, recipe_package: list[str] = Query(), user: U
...
 def get_lookup(application: str, user: User = Depends(auth_w_jwt_or_pat)):
-    """
-    Fetch a single lookup table.
+    """Get The Lookup Table for UI- and Storage Recipes the Provided Application
+
+    This endpoint fetches the recipe lookup table for the application provided.
+    This lookup table is used to find UI- and Storage recipes given a blueprint.
+
+    Args:
+        application (str): The name of the desired application.
```
Comment (KristianKjerstad) [inline src/features/lookup_table/lookup_table_feature.py:56]: add user argument?

**Review by KristianKjerstad — APPROVED** [review]
Good! Approved with some suggestions to consider

**Pull request merged**